themes/_core/js/checkout.js

```diff
@@ -51,9 +54,17 @@ function handleCheckoutStepChange() {
         $clickedStep.toggleClass('-current');
         $clickedStep.toggleClass(currentStepClass);
 
-        const $nextSteps = $clickedStep.nextAll();
-        $nextSteps.addClass('-unreachable').removeClass('-complete');
-        $('.step-title', $nextSteps).addClass('not-allowed');
+        if ($('button.continue', $clickedStep).length == 0) {
+          //If the step has no continue button, the previously completed steps are clickable
+          const $nextSteps = $clickedStep.nextAll('.checkout-step.-clickable');
+          $nextSteps.removeClass('-unreachable').addClass('-complete');
+          $('.step-title', $nextSteps).removeClass('not-allowed');
+        } else {
+          //If the step has a continue button, all next steps are disabled in order to force the user to click on continue
+          const $nextSteps = $clickedStep.nextAll();
+          $nextSteps.addClass('-unreachable').removeClass('-complete');
+          $('.step-title', $nextSteps).addClass('not-allowed');
+        }
```

Comment:
Consider extracting functions and objects, this code is very complicated to understand.

Picture this:

```js
if (!clickedStep.isUnreachable()) {
	clickedStep.makeCurrent();
	if (clickedStep.hasContinueButton()) {
		steps.disableAllAfter(clickedStep)
	} else {
        steps.enableAllBefore(clickedStep)
    }
}
```

Looks better isn't it?

Reply:
Yep much better, but then I need to extend jQuery functions? I'm not familiar with it but I can try
Or could this be enough:
```javascript
if (!isUnreachable(clickedStep)) {
	makeCurrent(clickedStep);
	if (hasContinueButton(clickedStep)) {
		disableAllAfter(steps, clickedStep)
	} else {
        	enableAllBefore(steps, clickedStep)
    	}
}

```